File: frontend/src/app/shared/drug/drug.ts
import {ATC} from "./atc";
import {DosageForm} from "./dosage-form";
import {ApplicationForm} from "./application-form";
import {NormPackageSize} from "./norm-package-size";
import {Producer} from "./producer";
import { SortMeta } from "primeng/api";

export interface DrugPageRequest {
    page?: number,
    size?: number,
    globalFilter?: string,
    multiSortMeta?: SortMeta[];
    sortField?: string,
    sortOrder?: number;
}

export interface DrugPageResponse {
    content: Drug[],
    totalElements: number,
    totalPages: number,
    size: number,
    number: number;
}

export class Drug {
    id: number;
    pharmaceuticalCentralNumber: number;
    atc: ATC;
    name: string;
    dosageForm: DosageForm;
    applicationForm: ApplicationForm;
    packageSize: number;
    normPackageSize: NormPackageSize;
    producer: Producer;
    priceInCents: number;
    fixedPriceInCents: number;
    isCustom = true;
}

File: backend/src/main/java/de/dzd/medlog/drug/Drug.java
package de.dzd.medlog.drug;

import de.dzd.medlog.drug.association.*;
import jakarta.persistence.*;
import lombok.*;

/**
 * stamm.txt
 */
@Entity
@Data
@RequiredArgsConstructor
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Drug {
    @Id
    @NonNull
    private Long pharmaceuticalCentralNumber;
    @ManyToOne
    private ATC atc;
    private String name;
    @ManyToOne
    private DosageForm dosageForm;
    @ManyToOne
    private ApplicationForm applicationForm;
    private Integer packageSize;
    @ManyToOne
    private NormPackageSize normPackageSize;
    @ManyToOne
    private Producer producer;
    private long priceInCents;
    private long fixedPriceInCents;
    private Boolean isCustom;
}
